cerbero/build/cookbook.py

```diff
@@ -420,21 +420,24 @@ def _load_recipes_from_dir(self, repo):
             # which can contain private classes to extend cerbero's recipes
             # and reuse them in our private repository
             try:
-                recipe = self._load_recipe_from_file(f, custom)
+                recipes_from_file = self._load_recipes_from_file(f, custom)
             except RecipeNotFoundError:
                 m.warning(_("Could not found a valid recipe in %s") % f)
-            if recipe is None:
+            if recipes_from_file is None:
                 continue
-            recipes[recipe.name] = recipe
+            for name in recipes_from_file:
+                recipes[name] = recipes_from_file[name]
         return recipes
 
-    def _load_recipe_from_file(self, filepath, custom=None):
+    def _load_recipes_from_file(self, filepath, custom=None):
+        recipes = {}
         mod_name, file_ext = os.path.splitext(os.path.split(filepath)[-1])
```

Comment:
I'm not sure to understand the difference with what i have changed...Do you want me to do the loop earlier as it was done before ?